finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4DispatcherBase.scala

```diff
@@ -105,6 +105,14 @@ trait Netty4DispatcherBase[SendMsg <: Message, RecvMsg <: Message] {
         log.debug(e, "[%s] dispatcher closed", prefix)
         Future.exception(e)
 
+      case Throw(e: ChannelClosedException)
+      // if all streams have already been closed, then this just means that
+      // the client failed to send a GOAWAY frame...
+      if closed.get || streams.keys.asScala.forall { _ <= closedId.get } =>
+        // ...so we don't need to propagate the exception
+        log.warning("[%s] client closed connection without sending GOAWAY frame", prefix)
```

Comment:
I'm torn about this.  Conceptually, I agree with you.  However, since some of the most widely used clients have this bad behavior, I worry about linkerd's logs just getting clogged with these messages.